src/mo_frontend/coverage.ml

```diff
@@ -300,16 +302,19 @@ and match_tup ctxt descs_r descs pats ts sets =
     match_pat (InTup (ctxt, descs_r, descs', pats', ts')) desc pat t sets
   | _ ->
     assert false
-    
+
 and match_obj ctxt ldescs (pat_fields : pat_field list) tfs sets =
   match pat_fields with
   | [] -> succeed ctxt (Obj ldescs) sets
-  | pat_field::pat_fields' ->
-    let l = pat_field.it.id.it in
+  | {it = VarPF(id, p); _}::pat_fields' ->
+    let l = id.it in
+    (* TODO: This seems wrong even before the addition of typed pattern fields? *)
     let tf = List.find (fun tf -> tf.T.lab = l) tfs in
```

Comment:
Yeah, I think this should be using Type.lookup_val_field or similar, otherwise it might find a type field of the same name looking for a value field.